feat(workflows): add reusable contracts deployment workflow

chriszhao1988 opened the pull request:
Adds a new GitHub Actions workflow for compiling and deploying smart contracts across multiple networks with configurable inputs and secrets.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a reusable contract deployment workflow that supports multiple networks, restores/saves deployment state, uploads ABIs, and updates deployed addresses.
  * Added DingTalk notification integration for deployment status alerts.

* **Chores**
  * Infrastructure updates for automated contract deployment pipeline and minor repository ignore rules.

<!-- review_stack_entry_start -->

[![Review Change Stack](https://storage.googleapis.com/coderabbit_public_assets/review-stack-in-coderabbit-ui.svg)](https://app.coderabbit.ai/change-stack/treasurenetprotocol/reusable-workflows/pull/56?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

<!-- review_stack_entry_end -->
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

chriszhao1988 committed:
feat(workflows): add reusable contracts deployment workflow

Adds a new GitHub Actions workflow for compiling and deploying smart contracts across multiple networks with configurable inputs and secrets.

chriszhao1988 committed:
chore(reusable-contracts-deploy.yml): refactor workflow inputs and add ABI upload to S3
- Replace verbose input/secrets definitions with concise inline syntax
- Remove network-specific RPC/private key logic; use direct secret injection
- Add jq installation, ABI JSON filtering, and S3 upload step for contract ABIs
- Set default S3 bucket and AWS region in environment variables

chriszhao1988 committed:
feat(reusable-contracts-deploy): add aws_region input and sm_secret_id support

- Added optional aws_region input to configure AWS region dynamically
- Added sm_secret_id secret to store contract addresses in Secrets Manager
- Updated AWS_REGION env to use inputs.aws_region fallback
- Added step to extract and upload latest contract addresses to Secrets Manager

chriszhao1988 committed:
chore(reusable-contracts-deploy): refactor workflow inputs and secrets to multi-line YAML format

dongchenxie commented on .github/workflows/reuseable-contracts-deploy.yml:
```diff
@@ -0,0 +1,145 @@
+      network: { required: true, type: string }
+    secrets:
+      rpc: { required: true }          # 调用方传入选定网络的 RPC
+      private_key: { required: true }  # 调用方传入选定网络的私钥
```
@chriszhao1988 please address

dongchenxie commented on .github/workflows/reuseable-contracts-deploy.yml:
```diff
@@ -0,0 +1,145 @@
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: npm
+          cache-dependency-path: package-lock.json
+
+      - name: Remove lockfile and node_modules
+        run: |
+          rm -f package-lock.json
+          rm -rf node_modules
+
+      - name: Install dependencies (fresh lock on runner OS)
+        run: npm install --no-audit --no-fund
```
@chriszhao1988 please address

chriszhao1988 committed:
fix(reusable-contracts-deploy.yml): use CONTRACT_DEPLOYER credentials and dynamic secret ID
Replace S3_OWNER credentials with CONTRACT_DEPLOYER credentials for AWS access. Update Secrets Manager secret ID to use dynamic network-based path.

chriszhao1988 committed:
refactor(reusable-contracts-deploy.yml): replace deprecated AWS secrets with generic access key inputs
- Updated workflow inputs to use `access_key_id` and `secret_access_key` instead of `sm_secret_id`
- Updated AWS environment variables in S3 upload and Secrets Manager steps to reference new secret names

chriszhao1988 committed:
feat(reusable-contracts-deploy): add S3 state persistence for deployments

- Restore deployments state from S3 before deployment
- Upload updated deployments state to S3 after deployment

chriszhao1988 committed:
fix(reusable-contracts-deploy): update S3 paths and flatten ABI uploads
- Changed deployment state path from /deployments/{NETWORK}.json to /{NETWORK}/state.json
- Flattened ABI JSON files before upload to avoid nested S3 paths
- Added timestamped and latest ABI uploads under /{NETWORK}/abis/

chriszhao1988 committed:
feat(reuseable-contracts-deploy.yml): add DingTalk notification for contract deployments
Add DingTalk webhook integration to notify on deployment status, including action, target, network, and deployed contracts summary.

chriszhao1988 committed:
refactor(workflows): extract DingTalk notification into reusable action

Split DingTalk notification logic from deploy workflow into a dedicated composite action to improve reusability and maintainability. The workflow now builds the message in one step and delegates sending to the new action.

chriszhao1988 committed:
feat(workflows): add reusable contract script workflow

Adds a new GitHub Actions workflow to run Hardhat scripts with configurable network and secrets.

chriszhao1988 committed:
refactor(reusable-contracts-deploy): update dingtalk-notify action path
Remove deprecated reusable-contracts-run-script.yml workflow and update dingtalk-notify action to use remote path.

chriszhao1988 committed:
fix: stop contract secret overwrite INF-119

chriszhao1988 committed:
fix: remove legacy contract state secret path INF-120

chriszhao1988 committed:
Merge pull request #58 from treasurenetprotocol/fix/contracts-secret-overwrite-INF-119

fix: stop contract secret overwrite INF-119

chriszhao1988 committed:
Merge pull request #61 from treasurenetprotocol/feature/remove-legacy-secret-path-contracts-INF-120

INF-120: remove legacy contract state secret path
